src/LayoutManager/ActiveSelectionLayoutManager.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { FabricObject } from '../shapes/Object/FabricObject';
+import { ActiveSelectionLayoutManager } from './ActiveSelectionLayoutManager';
+
+describe('ActiveSelectionLayoutManager', () => {
+  describe('onBeforeLayout', () => {
+    describe('triggers', () => {
+      const triggers: ('modified' | TModificationEvents | 'changed')[] = [
+        'modified',
+        'moving',
+        'resizing',
+        'rotating',
+        'scaling',
+        'skewing',
+        'changed',
+        'modifyPoly',
+      ];
+
+      it('should subscribe activeSelection that contains object', () => {
+        const manager = new ActiveSelectionLayoutManager();
+        const object = new FabricObject();
+        const group = new Group([object], {
+          interactive: true,
+          subTargetCheck: true,
+        });
+        const as = new ActiveSelection([object], { layoutManager: manager });
+        const objectOn = jest.spyOn(object, 'on');
+        const objectOff = jest.spyOn(object, 'off');
+        const asOn = jest.spyOn(as, 'on');
+        manager.subscribeTargets({
+          targets: [object],
+          target: as,
+        });
+        expect(objectOn).not.toHaveBeenCalled();
+        expect(objectOff).not.toHaveBeenCalled();
+        expect(asOn).toHaveBeenCalledTimes(triggers.length);
+        expect(objectOff).not.toHaveBeenCalled();
+      });
+
+      it('a subscribed activeSelection should trigger layout on the object parent once per parent', () => {
+        const manager = new ActiveSelectionLayoutManager();
+        const object = new FabricObject();
+        const object2 = new FabricObject();
+        const object3 = new FabricObject();
+        const object4 = new FabricObject();
+        const group = new Group([object, object2], {
+          interactive: true,
+          subTargetCheck: true,
+        });
+        const group2 = new Group([object3, object4], {
+          interactive: true,
+          subTargetCheck: true,
+        });
+        const as = new ActiveSelection([object, object2, object3, object4], {
+          layoutManager: manager,
+        });
+        const asPerformLayout = jest.spyOn(manager, 'performLayout');
+        const groupPerformLayout = jest.spyOn(
+          group.layoutManager,
+          'performLayout'
+        );
+        const groupPerformLayout2 = jest.spyOn(
+          group2.layoutManager,
+          'performLayout'
+        );
+        groupPerformLayout.mockClear();
+        groupPerformLayout2.mockClear();
+        asPerformLayout.mockClear();
+        expect(group.layoutManager['_subscriptions'].get(as)).toBeDefined();
+        expect(group2.layoutManager['_subscriptions'].get(as)).toBeDefined();
+        expect(manager['_subscriptions'].size).toBe(0);
+
+        const event = { foo: 'bar' };
+        triggers.forEach((trigger) => as.fire(trigger, event));
+        expect(asPerformLayout).not.toHaveBeenCalled();
+        expect(groupPerformLayout.mock.calls).toMatchObject([
```

Comment:
I wanted to do the same for `groupPerformLayout2` but it throws a weird error or max call stack exceeded

Reply:
It happened to me too, and there was a bug in the code in that moment. Maybe is worth double checking

Reply:
still happening

Reply:
Ok i ll look with fresh eyes, maybe there is a situation in which you get an infinite loop.

Reply:
no issues for me, it works fine

Reply:
I think you should have committed it

Reply:
i did i pushed up commits with the checks for groupPerformLayout2 mirrored to groupPerformLayout

Reply:
https://github.com/fabricjs/fabric.js/pull/9651/commits/a22cc7c17ee52a35f1d0f98a14d763ada235daf0

Reply:
Missed that! Great!